### DISCUSSION TITLE
Number of hyp3+mintpy interferograms

### DISCUSSION THREAD

**Discussion by DD-ZHAO** [insarlab/MintPy#1285] [Q&A] [answered]
Hello everyone, I recently used ASF Vertex's hyp3 to generate 79 interferograms, as shown in the figure below, and used Mintpy to perform time-series analysis on the 79 interferograms. 
![Figure1](https://github.com/user-attachments/assets/5e4d7e27-0929-4283-a9fa-eef165dd0f53)
However, without any network modifications, Mintpy only performed time-series analysis on 31 interferograms and there were gaps in the network, as shown in the following two figures. How can I solve this problem. Thank you everyone.
<img width="664" alt="Figure3" src="https://github.com/user-attachments/assets/dc000e4c-69fb-4f39-aa04-4f22cfe36392">
<img width="990" alt="Figure2" src="https://github.com/user-attachments/assets/513a35d0-7c36-45eb-a98d-9bb9ffe2c9b2">

**Comment by jhkennedy** [comment]
@DD-ZHAO to help, I'll need a little more information: 
* Can you share the log file, or the network selection part of the log? Usually the log file explains why interferograms were dropped from the network.
* What is the (geometric) reference scene you're using to pick the SBAS stack? 
* Have you looked into modifying your network? You can use `modify_network.py` to experiment with different selection criteria and `plot_network.py` to see how it affects the network.

**Reply by DD-ZHAO** [reply]
@jhkennedy Thank you very much for your reply！I have checked the log files and found that there are 48 interferograms that do not have a common size. There are a total of 31 interferograms with the most common size (2953 × 3641). The number of rows and columns for the remaining 48 interferograms is slightly different from the most common size. I think this is the reason why 79 interferograms were input while only 31 interferograms were used for time series analysis. The log file is shown in the following figure.
<img width="367" alt="mintpybug" src="https://github.com/user-attachments/assets/6253c20b-b2c7-4cea-b8e6-1a2eca6067a2">
Due to the fact that my interferograms were generated based on the hyp3 of ASF Vertex, and the images were obtained from the same path and frame, there may still be differences in the number of pixels. Can I avoid this problem by using AOI to crop the interferograms without processing the entire image? Thank you very much for your time!

**Reply by jhkennedy** [reply, marked as answer]
@DD-ZHAO, yes, we recommend subsetting all GeoTIFFs to their common overlap; see section 3.1 of our HyP3 + MintPy tutorial:
https://nbviewer.org/github/ASFHyP3/hyp3-docs/blob/main/docs/tutorials/hyp3_insar_stack_for_ts_analysis.ipynb

Note: you may have to be careful when selecting images for submitting -- sometimes there is enough drift in the SLC frames, or if your AOI is near frame boundaries, you can end up with extremely small common areas.

**Reply by DD-ZHAO** [reply]
@jhkennedy Thank you very much for your prompt reply! I will try to only process interferograms cropped to AOI.
